### PR TITLE
Adding IoT section

### PR DESCRIPTION
**Pull request by lynnlangit**
First PR on new IoT section.  NOTE: I did NOT update the README.md index to add the links for the new IoT section yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1308,6 +1306,59 @@ This section covers tips and information on achieving [high availability](https:
...
 
+IoT
+---------------------------
+
+
+### IoT Basics
+
+* 📒 [Homepage](https://aws.amazon.com/iot/) ∙ [User guide](https://docs.aws.amazon.com/iot/latest/developerguide/) ∙ [FAQ](https://aws.amazon.com/iot/faqs/) ∙ [Pricing](https://aws.amazon.com/iot/pricing/)
+* **IoT** is a platform for allowing clients such as IoT devices or software applications ([examples](http://internetofthingswiki.com/iot-applications-examples/541/)) to communicate with the AWS cloud.
+* Clients are also called **devices** (or **things**) and include a wide variety of device types.  Roughly there are three categories of device types that interact with IoT services by sending message over an IoT protocol to the IoT Pub/Sub-style message broker, which is called the IoT** Device Gateway**:
```
Comment (jlevy) [inline README.md]: Typo with ** boldface

File: README.md [hunk 2]
```diff
@@ -1308,6 +1306,59 @@ This section covers tips and information on achieving [high availability](https:
...
+    * AWS [**IoT Policies**](http://docs.aws.amazon.com/iot/latest/developerguide/authorization.html) (device/topic authorization) are JSON files that are associated to one or more AWS IoT certificates. This authorizes associated devices to publish and/or subscribe to messages from one or more MQTT topics.
+    * AWS [**IoT Rules**](http://docs.aws.amazon.com/iot/latest/developerguide/iot-rules.html) are SQL-like queries which allows for reuse of some or all device message data, as described in [this presentation, which summarizes design patterns with for IoT Rules](http://www.slideshare.net/AmazonWebServices/programming-the-physical-world-with-device-shadows-and-rules-engine-66486454).
+    * Shown below is a [diagram](https://aws.amazon.com/iot/how-it-works/) which summarizes the flow of messages between the AWS IoT services:
+    * [Image: https://d0.awsstatic.com/IoT/diagrams/awsiot-how-it-works_HowITWorks_1-26.png]
```
Comment (jlevy) [inline README.md]: I don't think this syntax works. Can you match the other syntax and click "view" to see it is showing up? Will also want to drop the bullet at the front.

**Comment by jlevy** [comment]
@ThanosBaskous and other IoT experts might want to do a final pass too.

@ThanosBaskous note I think we can do the markdownfmt consistency once we are caught up on PRs to minimize conflicts. (But go ahead and argue w me if you disagree.)

File: README.md (hunk 1 above)
Comment (lynnlangit) [inline README.md]: what is the line number, or item with the typo?

File: README.md (hunk 2 above)
Comment (lynnlangit) [inline README.md]: Sorry I am not clear on what you want me to do - what does 'match the other syntax mean'?  What does 'drop the bullet at front' mean?  line number?

File: README.md [hunk 3]
```diff
@@ -1308,6 +1306,59 @@ This section covers tips and information on achieving [high availability](https:
 -	**AZ naming** differs from one customer account to the next. Your “us-west-1a” is not the same as another customer’s “us-west-1a” — the letters are assigned to physical AZs randomly per account. This can also be a gotcha if you have multiple AWS accounts.
 -	**Cross-AZ traffic** is not free. At large scale, the costs add up to a significant amount of money. If possible, optimize your traffic to stay within the same AZ as much as possible.  
 
+IoT
+---------------------------
+
```
Comment (ThanosBaskous) [inline README.md]: nit: one newline between section header and sections' first subsection

File: README.md (hunk 1 above)
Comment (ThanosBaskous) [inline README.md]: style nit: 1 space before new sentence.

File: README.md [hunk 4]
```diff
@@ -1308,6 +1306,59 @@ This section covers tips and information on achieving [high availability](https:
...
+* AWS has a useful [quick-start](http://docs.aws.amazon.com/iot/latest/developerguide/iot-gs.html) (using the Console) and a [slide presentation](http://www.slideshare.net/AmazonWebServices/connecting-to-aws-iot) on core topics.
+* **IoT terms:**
+    * AWS [**IoT Things**](http://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html) (metadata for devices in a [registry](http://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html)) and can store device state in a JSON document, which is called a [**device shadow**](http://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-shadows.html).
+    * AWS [**IoT Certificates**](http://docs.aws.amazon.com/iot/latest/developerguide/attach-cert-thing.html) (device authentication) are the logical association of a unique certificate to the logical representation of a device. This association can be done in the Console.  In addition the public key of the certificate must be copied to the physical device. This covers the authentication of devices to a particular AWS Device Gateway (or message broker).
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: "In addition, "

File: README.md [hunk 5]
```diff
@@ -1308,6 +1307,59 @@ This section covers tips and information on achieving [high availability](https:
...
+* Issues of lock-in center around your devices —  [protocols](http://www.postscapes.com/internet-of-things-protocols/) (for example MQTT, AMQP), message formats (such as, JSON vs. Hex...) and security (certificates).
+
+### IoT Tips
+
+* **Getting started with Buttons:** One way to start is to use an [**AWS IoT Button**](https://aws.amazon.com/iot/button/).  AWS provides a number of code samples for use with their IoT Button, you can use the AWS IoT console, click the “connect AWS IoT button” link and you'll be taken to the  AWS Lambda console.  There you fill out your button’s serial number to associate it with a Lambda. (As of this writing, AWS IoT buttons are only available for sale in the US.)
+* **Connections and protocols:** It is important to understand detail about the devices you wish to connect to the AWS IoT service, including how you will secure the device connections, the device protocols, and more. Cloud vendors differ significantly in their support for common IoT protocols, such as MQTT, AMQP, XMPP. AWS IoT supports **secure MQTT**, **WebSockets** and **HTTPS**.
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: detail -> the details of

File: README.md [hunk 6]
```diff
@@ -1308,6 +1307,59 @@ This section covers tips and information on achieving [high availability](https:
...
+
+* **IoT protocols:** It is important to verify the exact type of support for your particular IoT device message protocol. For example, one commonly used IoT protocol is [MQTT](https://www.ibm.com/developerworks/community/blogs/5things/entry/5_things_to_know_about_mqtt_the_protocol_for_internet_of_things?lang=en). Within MQTT there are [three possible levels of QoS in MQTT](https://dzone.com/articles/internet-things-mqtt-quality).  AWS IoT supports MQTT [QoS 0](http://docs.aws.amazon.com/iot/latest/developerguide/protocols.html) (fire and forget, or at most once) and QoS 1(at least once, or includes confirmation), but *not* QoS 2 (exactly once, requires 4-step confirmation).  This is important in understanding how much code you’ll need to write for your particular application message resolution needs.  Here is a [presentation about the nuances of connecting](http://www.slideshare.net/AmazonWebServices/overview-of-iot-infrastructure-and-connectivity-at-aws-getting-started-with-aws-iot).
+* The ecosystems to match **IAM users or roles** to **IoT policies** and their associated authorized AWS IoT devices is immature. Custom coding to enforce your security requirements is common.
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: either "ecosystems .. are immature" or "ecosystem .. is immature" (the latter seems more corret to me)

**Comment by ThanosBaskous** [comment]
@jlevy Good note regarding markdown consistency - I'll ease up on nitpicky markdown formatting issues.

File: README.md (hunk 1 above)
Comment (lynnlangit) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (lynnlangit) [inline README.md]: done

File: README.md (hunk 5 above)
Comment (lynnlangit) [inline README.md]: done

File: README.md (hunk 6 above)
Comment (lynnlangit) [inline README.md]: done

**Comment by lynnlangit** [comment]
I believe I have updated to include all of both of your change requests.

File: README.md [hunk 7]
```diff
@@ -1308,6 +1307,58 @@ This section covers tips and information on achieving [high availability](https:
...
+- 	The ecosystems to match **IAM users or roles** to **IoT policies** and their associated authorized AWS IoT devices are immature. Custom coding to enforce your security requirements is common.
+- 	A common mistake is to misunderstand the importance of IoT **device** **security**.  It is imperative to associate *each* device with a unique certificate (public key). You can generate your own certificates and upload them to AWS, or you can use AWS generated IoT device certificates. It’s best to read and understand AWS’s own guidance on this [topic](http://www.slideshare.net/AmazonWebServices/best-practices-of-iot-in-the-cloud).
```
Comment (jlevy) [inline README.md]: Is this one so big it should have a ❗?

File: README.md [hunk 8]
```diff
@@ -1308,6 +1307,58 @@ This section covers tips and information on achieving [high availability](https:
...
+    * 	Use this [MQTT load simulator](https://github.com/awslabs/aws-iot-mqtt-load-generator) to test device message load throughout your IoT solution.
+
+### IoT Gotchas and Limitations
+
+- 	**IoT protocols:** It is important to verify the exact type of support for your particular IoT device message protocol. For example, one commonly used IoT protocol is [MQTT](https://www.ibm.com/developerworks/community/blogs/5things/entry/5_things_to_know_about_mqtt_the_protocol_for_internet_of_things?lang=en). Within MQTT there are [three possible levels of QoS in MQTT](https://dzone.com/articles/internet-things-mqtt-quality).  AWS IoT supports MQTT [QoS 0](http://docs.aws.amazon.com/iot/latest/developerguide/protocols.html) (fire and forget, or at most once) and QoS 1(at least once, or includes confirmation), but *not* QoS 2 (exactly once, requires 4-step confirmation).  This is important in understanding how much code you’ll need to write for your particular application message resolution needs.  Here is a [presentation about the nuances of connecting](http://www.slideshare.net/AmazonWebServices/overview-of-iot-infrastructure-and-connectivity-at-aws-getting-started-with-aws-iot).
```
Comment (jlevy) [inline README.md]: Probably all of these should have a 🔸 at the front (unless the security one is so big it deserves a ❗).

**Comment by jlevy** [comment]
LGTM now, with just two details:
- The gotchas are missing the emoji yellow🔸 or ❗
- There is a line in https://github.com/open-guides/og-aws#which-services-to-use 

⛓IoT: Manage bidirectional communication over HTTP, WebSockets, and MQTT between AWS and clients (often but not necessarily “things” like appliances or sensors)

Could we change the link there to be to #iot now, and are you good with that summary too btw?

File: README.md (hunk 8 above)
Comment (lynnlangit) [inline README.md]: done

File: README.md (hunk 7 above)
Comment (lynnlangit) [inline README.md]: done

**Comment by ThanosBaskous** [comment]
💥 Awesome! Thanks for this fantastic addition @lynnlangit!

**Comment by jlevy** [comment]
👍 👍 👍 Merged last changes in (and moved section to final place).
Awesome. Thank you @lynnlangit !

**Pull request merged**